polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -490,26 +490,16 @@ impl<T: Config> Pallet<T> {
 		// Contains the disputes that are concluded in the current session only,
 		// since these are the only ones that are relevant for the occupied cores
 		// and lightens the load on `free_disputed` significantly.
-		// Cores can't be occupied with candidates of the previous sessions, and only
-		// things with new votes can have just concluded. We only need to collect
-		// cores with disputes that conclude just now, because disputes that
-		// concluded longer ago have already had any corresponding cores cleaned up.
-		let current_concluded_invalid_disputes = checked_disputes_sets
-			.iter()
-			.map(AsRef::as_ref)
-			.filter(|dss| dss.session == current_session)
-			.map(|dss| (dss.session, dss.candidate_hash))
-			.filter(|(session, candidate)| {
-				<T>::DisputesHandler::concluded_invalid(*session, *candidate)
-			})
-			.map(|(_session, candidate)| candidate)
-			.collect::<BTreeSet<CandidateHash>>();
+		let mut current_concluded_invalid_disputes =
+			<T>::DisputesHandler::disputes_concluded_invalid(current_session);
```

Comment:
Compared to before, there can be quite a lot of disputes returned if there was a dispute storm, but we never iterate. 

However search is `O(log n)` and n is no longer bounded by how many you can put in the inherent, but I have no idea how if it would affect the weight significantly.